ansible_collections/arista/avd/plugins/plugin_utils/cv_client/workflows/verify_devices_on_cv.py

```diff
@@ -22,11 +24,59 @@ async def verify_devices_on_cv(devices: list[CVDevice], workspace_id: str, skip_
     if not devices:
         return
 
+    # Finding entries with same serialNum or same mac address
+    sysMacToHostname = {}
+    serialNumToHostname = {}
+    for e in devices:
+        if e.system_mac_address:
+            if e.system_mac_address in sysMacToHostname:
+                sysMacToHostname[e.system_mac_address].append(e.hostname)
+            else:
+                sysMacToHostname[e.system_mac_address] = [e.hostname]
+
+        if e.serial_number:
+            if e.serial_number in serialNumToHostname:
+                serialNumToHostname[e.serial_number].append(e.hostname)
+            else:
+                serialNumToHostname[e.serial_number] = [e.hostname]
+    hostnamesSameMac = [hostnames for mac, hostnames in sysMacToHostname.items() if len(hostnames) > 1]
+    hostnames_with_same_serial = [hostnames for mac, hostnames in serialNumToHostname.items() if len(hostnames) > 1]
+    if hostnamesSameMac:
+        raise CVDuplicateInfo("Same System MAC found for ", hostnamesSameMac)
+    if hostnames_with_same_serial:
+        raise CVDuplicateInfo("Same Serial Number found for ", hostnames_with_same_serial)
+
     existing_devices = await verify_devices_in_cloudvision_inventory(devices, skip_missing_devices, warnings, cv_client)
     await verify_devices_in_topology_studio(existing_devices, workspace_id, cv_client)
     return
 
 
+def return_duplicates(found_device_dict_by_serial: dict, found_device_dict_by_system_mac: dict, found_devices: list[Device]) -> Exception:
```

Comment:
The code in here is duplicate of the code above. Please work out a good arrangement of the code, where the flow is easy to follow and we avoid deep indentations and repetitions.

Reply:
You could move this input validation into a function like `validate_input` or similar. Also remember docstring to describe what you are doing in the the function.